### PR TITLE
Resize "Offers to buy/sell with X"; list to Bisq's vertical window size

### PR DESCRIPTION
**Pull request by ghubstan**
Replaces closed PR 1373 (had dirty commit history).

### CODE REVIEW

**Review by cbeams — APPROVED** [review]
utACK.

@ManfredKarrer please review as well, I've only looked at this for PR correctness, not its substance.

@ghubstan, in future commits, please:

Follow the commit comment guidelines at https://chris.beams.io/posts/git-commit.

And In GitHub PRs and issues, please link to issues with hashmarks, e.g. #1373 (your description in this PR mentioned the issue number without the `#`). Doing this creates linkage and discoverability between the issues in the GitHub UI.

Thanks!

File: gui/src/main/java/io/bisq/gui/main/market/offerbook/OfferBookChartView.java [hunk 1]
```diff
@@ -88,7 +89,13 @@
     private HBox bottomHBox;
     private ListChangeListener<OfferBookListItem> changeListener;
     private ListChangeListener<CurrencyListItem> currencyListItemsListener;
-
+    private ChangeListener<Number> bisqWindowVerticalSizeListener;
+    private final double pixelsPerOfferTableRow = (109.0 / 4.0) + 10.0;
+    private final Function<Double, Double> offerTableViewHeight = (screenSize) -> {
+        // startup defaults: scene height = 710   tbl view height = 109   visible row count = 4
+        int extraRows = screenSize <= 710 ? 0 : (int) ((screenSize - 710) / pixelsPerOfferTableRow);
```
Comment (ripcurlx) [inline gui/src/main/java/io/bisq/gui/main/market/offerbook/OfferBookChartView.java]: Maybe you could move the initial layout height (710) as a field to the [Layout class](https://github.com/bisq-network/exchange/blob/master/gui/src/main/java/io/bisq/gui/util/Layout.java) and also replace it in the [BisqApp class](https://github.com/bisq-network/exchange/blob/1d0591084d6d2cc8204357204ca0f54b887ab765/gui/src/main/java/io/bisq/gui/app/BisqApp.java#L235) so the code will work as expected if we decide to change the default height at some point. Also please move the table view height (109) in a field and re-use it in the table view initialization https://github.com/bisq-network/exchange/blob/1d0591084d6d2cc8204357204ca0f54b887ab765/gui/src/main/java/io/bisq/gui/main/market/offerbook/OfferBookChartView.java#L302

**Review by ripcurlx — CHANGES_REQUESTED** [review]
Besides moving the numbers into fields everything else works as expected 👍

File: gui/src/main/java/io/bisq/gui/main/market/offerbook/OfferBookChartView.java (hunk 1 above)
Comment (ghubstan) [reply gui/src/main/java/io/bisq/gui/main/market/offerbook/OfferBookChartView.java]: I pushed the changes you asked for in two commits that should have been one.

**Review by ghubstan — COMMENTED** [review]
Review request for changes made as per

"Maybe you could move the initial layout height (710) as a field to the Layout class and also replace it in the BisqApp class so the code will work as expected if we decide to change the default height at some point. Also please move the table view height (109) in a field and re-use it in the table view initialization "

**Review by ripcurlx — APPROVED** [review]

**Comment by ripcurlx** [comment]
ACK

**Pull request merged**